pyV2DL3/vegas/fillEVENTS_not_safe.py

```diff
@@ -80,57 +79,75 @@ def __fillEVENTS_not_safe__(vegasFileIO, event_classes=None, save_msw_msl=False)
     logger.debug("Start filling events ...")
 
     for ev in selectedEventsTree:
-        # When not using event class mode, this will stay 0
-        event_class_idx = 0
+        # Event reconstruction method
+        if reco_type == 1:
+            reco = ev.S
+        elif reco_type == 2:
+            reco = ev.M3D
+        else:
+            raise Exception("Invalid reconstruction type!"
+                            "\nSee --help for supported arguments")
+
+        # Include event in averages regardless of whether it gets cut
+        avAlt.append(reco.fArrayTrackingElevation_Deg)
+        avAz.append(reco.fArrayTrackingAzimuth_Deg)
+        avRA.append(reco.fArrayTrackingRA_J2000_Rad)
+        avDec.append(reco.fArrayTrackingDec_J2000_Rad)
 
-        if event_classes is not None:
-            fMSW = ev.S.fMSW
+        event_class_idx = 0
+        if event_class_mode:
+            fMSW = reco.fMSW
             """Determine which event class (if any) the event falls into.
 
-            Simply loop through the event classes and break if this event meets all
-            of an event class' parameters
+            Simply loop through the effective area files and break if this event meets all
+            of one's cuts parameters
 
             For now, we only do it based on the MSW intervals
             """
-            event_class_idx = 0
-            # Loop through event classes to check if this event satisfies the parameters
-            for ec in event_classes:
+            for ec in effective_area_files:
                 if ec.msw_lower <= fMSW < ec.msw_upper:
                     break
                 event_class_idx += 1
 
             # If this event falls into an event classes
             if event_class_idx < num_event_groups:
                 event_groups[event_class_idx]["mswArr"].append(fMSW)
```

Comment:
Difficult for me to understand all the logic of the VEGAS part - but noticed that here you append `fMSW`, while in the following lines it is always `reco.fMSW`. Please cross check that this is ok.

Reply:
Yeah this part is hard to read, I assign fMSW = reco.fMSW since it is accessed a few times for each event (of which there are thousands). I believe this is somewhat more efficient since the reconstructed event tree has many different data members

Reply:
Hi @tmcahill, yes its tricky to read here in the review pane, but worth double checking that there is no danger of fMSW being undefined in the final append statement on line 127? It looks as though you assign fMSW = reco.fMSW on line 99 inside an if statement, which appears at an inner level relative to the append statement. Perhaps I missed a previous declaration of fMSW or it's just the way things appear in the review pane for me, but if you can just double check this that would be good.

Reply:
Hi @tmcahill, yes its tricky to read here in the review pane, but worth double checking that there is no danger of fMSW being undefined in the final append statement on line 127? It looks as though you assign fMSW = reco.fMSW on line 99 inside an if statement, which appears at an inner level relative to the append statement. Perhaps I missed a previous declaration of fMSW or it's just the way things appear in the review pane for me, but if you can just double check this that would be good.

Reply:
Ah, I see. This is true for most languages. In python, one can assign and declare variables at the same time, and if statements do not constitute their own scope. The if condition used before assigning fMSW is identical to the condition which uses it. I could not pair them under the same if statement because the FoV check must happen before assigning fMSW to the event list but after finding which event class the event falls into.